### ISSUE TITLE
Unrecognized tensor type ID: AutocastCUDA

### ISSUE THREAD

**Issue by fancyerii** [huggingface/optimum#1474]
### System Info

```shell
transformers version: 4.34.1
Platform: Linux-5.4.0-153-generic-x86_64-with-glibc2.31
Python version: 3.9.18
Huggingface_hub version: 0.17.3
Safetensors version: 0.4.0
Accelerate version: 0.23.0
optimum version: 1.13.2
```

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction (minimal, reproducible, runnable)

when I try [this tutorial](https://colab.research.google.com/drive/1_TIrmuKOFhuRRiTWN94iLKUFu6ZX4ceb?usp=sharing#scrollTo=vT0XjNc2jYKy).

File "/home/lili/miniconda3/envs/py39_torch21/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
return forward_call(*args, **kwargs)
File "/home/lili/miniconda3/envs/py39_torch21/lib/python3.9/site-packages/peft/tuners/lora.py", line 1255, in forward
result = self.quant_linear_module(x)
File "/home/lili/miniconda3/envs/py39_torch21/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
return self._call_impl(*args, **kwargs)
File "/home/lili/miniconda3/envs/py39_torch21/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
return forward_call(*args, **kwargs)
File "/home/lili/miniconda3/envs/py39_torch21/lib/python3.9/site-packages/auto_gptq/nn_modules/qlinear/qlinear_cuda_old.py", line 221, in forward
self.autogptq_cuda.vecquant4matmul_old(x, self.qweight, out, self.scales.float(), self.qzeros, self.group_size)
RuntimeError: Unrecognized tensor type ID: AutocastCUDA

Hardware details
Nvidia A100 40GB

Software
ubuntu 20.04 torch=2.1.0+cu118 auto-gptq=0.4.2+cu117 accelerate=0.23.0

I have found related issue in https://github.com/PanQiWei/AutoGPTQ/issues/374 but found no answer.

### Expected behavior

no error

**Comment by fxmarty** [comment]
Hi @fancyerii , it may be due to https://github.com/PanQiWei/AutoGPTQ/blob/e4b2493733d69a6e60e22cebc64b619be39feb0e/auto_gptq/modeling/_base.py#L452.

**Comment by fancyerii** [comment]
hi @fxmarty, Then how to fix it?

**Comment by fxmarty** [comment]
Are you using `generate`? Could you try removing the context manager for autocast?

**Comment by MatheusRDG** [comment]
Same issue here

`outputs = model.generate(**inputs, generation_config=generation_config)
`

The source code is here [zephyr finetuning](https://github.com/Vasanthengineer4949/NLP-Projects-NHV/blob/main/LLMs%20Related/Finetune%20Zephyr/Zephyr7B_Chatbot.ipynb)

**Comment by MatheusRDG** [comment]
Solved with downgrade of torch to 2.0.1.

**Comment by fxmarty** [comment]
Closing as duplicate of https://github.com/huggingface/optimum/issues/1474
